File: app/src/main/java/com/patientapp/health/data/UserRole.java
package com.patientapp.health.data;

public enum UserRole {
    DOCTOR,
    PATIENT;

    public static UserRole fromString(String value) {
        if (value == null) return null;
        try {
            return valueOf(value.toUpperCase());
        } catch (IllegalArgumentException e) {
            return null;
        }
    }
}

File: app/src/main/java/com/patientapp/health/data/UserRole.kt
package com.patientapp.health.data

enum class UserRole {
    DOCTOR,
    PATIENT;

    companion object {
        fun fromString(value: String?): UserRole? = when (value?.uppercase()) {
            "DOCTOR" -> DOCTOR
            "PATIENT" -> PATIENT
            else -> null
        }
    }
}
